data/integration_tests/percolator/IntegrationTest_Percolator_SpeedGraph.py

```diff
@@ -0,0 +1,167 @@
+# Marcus Andersson - Percolator Project
+# Takes data from IntegrationTest_Percolator_SpeedGraphData.py and outputs graph images.
+# Input: Any number of data files, but at least 1.
+
+pathToOutputData = "@pathToOutputData@"
+pathToTestScripts = "@pathToTestScripts@"
+
+import sys
+
+import pathlib
+import datetime
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import ast 
+from matplotlib.cm import get_cmap
+import matplotlib.lines as mlines
+from matplotlib.ticker import FormatStrFormatter
+
+
+markers = ["o","v","^","<",">","1","s","p","P","*","H","x","d","|","+"]
+outputFilename = "outGraph"
+
+def getMinExponent(xStepValue):
+    exponent = 1
+    while xStepValue % 10 == 0 and xStepValue > 0:
+        exponent *= 10
+        xStepValue /= 10
+    return exponent
+
+def addPlotData(data, name, y_values, x_values):
+    d = {}
+    for i in range (0, len(x_values)):
+        d[x_values[i]] = y_values[i]
+    data[name] = d
+
+def getPlotData(files, fileNames, xSteps):
+    dataWall = {}
+    dataCPU = {}
+    for i in range(0,len(files)):
+        f = open(files[i], "r")
+        lines = f.read().splitlines()
+        f.close()
+        runTimes = ast.literal_eval(lines[0]) 
+        wallSeconds = [item[1] for item in runTimes]
+        cpuSeconds = [item[0] for item in runTimes]
+        addPlotData(dataWall, fileNames[i], wallSeconds, xSteps)
+        addPlotData(dataCPU, fileNames[i], cpuSeconds, xSteps)
+    return dataWall, dataCPU
+
+def getPlotColors():
+    colorScheme = "Dark2"
+    cmap = get_cmap(colorScheme)  # type: matplotlib.colors.ListedColormap
+    return cmap.colors 
+
+def setBarPlotLegends(data, colors):
+    colorCounter = 0
+    legendColors = {}
+    for keyName in data.keys():
+        legendColors[keyName] = colors[colorCounter]
+        colorCounter += 1      
+    labels = list(legendColors.keys())
+    handles = [plt.Rectangle((0,0),1,1, color=legendColors[label]) for label in labels]
+    plt.legend(handles, labels)
+
+def setLinePlotLegends(data, colors):
+    counter = 0
+    legendColors = {}
+    legendMarkers = {}
+    for keyName in data.keys():
+        legendColors[keyName] = colors[counter]
+        legendMarkers[keyName] = markers[counter]
+        counter += 1      
+    labels = list(legendColors.keys())
+    handles = [plt.Line2D([0,0],[0,1], color=legendColors[label], marker=legendMarkers[label], linestyle='-') for label in labels]
+    plt.legend(handles, labels)
+
+def setDescriptiveText(minExponent, titleName):
+    plt.suptitle(titleName, fontsize='xx-large')
+    plt.xlabel('Number of PSM Lines x$10^{}$'.format(str(len(str(minExponent))-1)), fontsize='x-large')
```

Comment:
The outer str() is unnecessary -- format() automatically turns everything into a string.